### PR TITLE
AMP Support for Quick Search component

### PR DESCRIPTION
**Pull request by gaetanmarmasse**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | N/A
| Patch: Bug Fix?          | No
| Minor: New Feature?      | Yes, AMP support
| Major: Breaking Change?  | No
| Tests Added + Pass?      | Yes
| Documentation Provided   | Yes (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

This PR adds AMP support for the quick search component. It uses amp-form, amp-autocomplete, amp-bind and amp-mustache in order to provide the same functionality as the non-amp search component.
Note: It comes with a content policy attached to the components library sample site that will change the default settings of the search component so that it allows for 30 results as well as a trigger of the search query in the backend starting at 1 character instead of 3.
To test this component, build as usual and navigate to http://localhost:4502/content/core-components-examples/library/amp.amp.html?wcmmode=disabled in order to see the component working. (assuming you are in paired AMP or AMP only mode).

### CODE REVIEW

File: examples/src/content/jcr_root/apps/core-components-examples/components/search/customheadlibs.amp.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
+
+<script async custom-element="amp-autocomplete" src="https://cdn.ampproject.org/v0/amp-autocomplete-0.1.js"></script>
+<script async custom-element="amp-form" src="https://cdn.ampproject.org/v0/amp-form-0.1.js"></script>
+<script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
+<script async custom-element="amp-bind" src="https://cdn.ampproject.org/v0/amp-bind-0.1.js"></script>
```
Comment (paigekeenan) [inline examples/src/content/jcr_root/apps/core-components-examples/components/search/customheadlibs.amp.html]: I don't think we are using `bind` anymore and that we can remove the script include for it.

File: examples/src/content/jcr_root/apps/core-components-examples/components/search/amp.html [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          <!-- Autocomplete results container -->
+          <template type="amp-mustache" id="amp-template-custom">
+              <div data-value="{{ url }}">
+                  <a href="{{ url }}">{{ title }}</a>
+              </div>
+          </template>
+        </amp-autocomplete>
+    </div>
+  </form>
+<div class="cmp-search__results" data-cmp-hook-search="results" role="listbox" aria-multiselectable="false"></div>
```
Comment (paigekeenan) [inline examples/src/content/jcr_root/apps/core-components-examples/components/search/amp.html]: We shouldn't need `.cmp-search__results` anymore.

**Review by paigekeenan — APPROVED** [review]
LGTM! Just a couple of html call outs from me.

File: examples/src/content/jcr_root/apps/core-components-examples/components/search/customheadlibs.amp.html (hunk 1 above)
Comment (gaetanmarmasse) [reply examples/src/content/jcr_root/apps/core-components-examples/components/search/customheadlibs.amp.html]: Great catch! Removed.

**Pull request merged**